java-manta-client-unshaded/src/main/java/com/joyent/manta/http/StandardHttpHelper.java

```diff
@@ -366,28 +422,75 @@ public MantaObjectInputStream httpRequestAsInputStream(final HttpUriRequest requ
                 throw e;
             }
 
-            return new MantaObjectInputStream(metadata, response,
-                    backingStream);
-        };
-
-        final int expectedHttpStatus;
+            final InputStream backingStream;
+            final InputStreamContinuator continuator = constructContinuatorForCompatibleRequest(request, response);
+            if (continuator != null) {
+                backingStream = new AutoContinuingInputStream(httpEntityStream, continuator);
+            } else {
+                backingStream = httpEntityStream;
+            }
 
-        if (requestHeaders != null && requestHeaders.containsKey(HttpHeaders.RANGE)) {
-            expectedHttpStatus = HttpStatus.SC_PARTIAL_CONTENT;
-        } else {
-            expectedHttpStatus = HttpStatus.SC_OK;
-        }
+            return new MantaObjectInputStream(metadata, response, backingStream);
+        };
 
         return executeRequest(
                 request,
                 expectedHttpStatus,
                 responseAction,
-                false, "GET    {} response [{}] {} ");
+                false,
+                "GET    {} response [{}] {} ");
+    }
+
+    /**
+     * Attempts to construct a {@link InputStreamContinuator} for the initial request-response exchange. If the request
+     * cannot be resumed (either because the exchange is malformed or the request does not support resuming) we will
+     * return null, otherwise the continuator receives our client, the initial request (which it will clone) and the
+     * marker we created.
+     *
+     * @param request the initial request, etag and range headers will be used as hints for the first argument to
+     * {@link ResumableDownloadMarker#validateInitialExchange(Pair, int, Pair)}
+     * @param response the initial response which will be validated against any hints
+     * @return the continuator which can be used to resume this request
+     */
+    private InputStreamContinuator constructContinuatorForCompatibleRequest(final HttpUriRequest request,
+                                                                            final CloseableHttpResponse response) {
+        if (this.maxDownloadContinuations == DOWNLOAD_CONTINUATIONS_DISABLED
+                || !HttpGet.METHOD_NAME.equalsIgnoreCase(request.getMethod())
+                || !(this.connectionContext instanceof MantaApacheHttpClientContext)
+                || !(request instanceof HttpGet)) {
+            return null;
+        }
+
+        final HttpGet get = (HttpGet) request;
+        final ResumableDownloadMarker marker;
+        try {
+            marker = validateInitialExchange(
+                    extractDownloadRequestFingerprint(get),
+                    response.getStatusLine().getStatusCode(),
+                    extractDownloadResponseFingerprint(response, true));
+        } catch (final ProtocolException pe) {
```

Comment:
A note here about why the exception is being swallowed and null returned would be useful to the reader.

Reply:
Added a debug log which includes the reason we couldn't build a marker